calculator/src/main/java/com/market/banica/calculator/service/ProductServiceImpl.java

```diff
@@ -313,36 +306,47 @@ private String getProductName(List<Product> products) {
         return products.get(0).getProductName();
     }
 
-    private void addAllIngredientsFromProductInListAsProduct(List<Product> result, Product recipe) {
-        LOGGER.debug("In addAllIngredientsFromProductInListAsProduct private method");
+    private void addAllIngredientsFromProductToMapOfProductAndQuantity(Map<Product, Map<String, Long>> productQuantitiesMap,
+                                                                       Product parentProduct) {
+        LOGGER.debug("In addAllIngredientsFromProductToMapOfProductAndQuantity private method");
 
-        Queue<Product> tempContainer = convertListOfProductNamesInQueueOfProducts(recipe);
+        Queue<Product> tempContainer = new ArrayDeque<>();
+        tempContainer.add(parentProduct);
 
         while (!tempContainer.isEmpty()) {
 
-            Product tempProduct = tempContainer.remove();
+            Product tempParentProduct = tempContainer.remove();
 
-            if (!tempProduct.getIngredients().isEmpty()) {
+            if (!tempParentProduct.getIngredients().isEmpty()) {
```

Comment:
a null check on tempParentProduct.getIngredients can be added as well

Reply:
Fixed